api/segments/models.py

```diff
@@ -198,6 +198,38 @@ def deep_clone(self) -> "Segment":
 
         return cloned_segment
 
+    def match_rules_to_segment(self, segment: "Segment") -> None:
```

Comment:
This code is really hard to follow. I've added some specific suggestions below, but we definitely need a docstring here, especially as this code isn't even (seemingly at least) called from this repository.

Reply:
1. Yeah the code is being called from the workflows repository (where there is more testing logic there as well). 
2. I've added a docstring to this method.